File: 538-convert-bst-to-greater-tree/538-convert-bst-to-greater-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def solve(self, root):
        if root == None:
            return
        
        self.solve(root.right)
        
        self.valSum += root.val
        root.val = self.valSum
        
        self.solve(root.left)
        
        
    def convertBST(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        # Daily
        self.valSum = 0
        self.solve(root)
        
        return root
        

File: 538-convert-bst-to-greater-tree/538-convert-bst-to-greater-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* convertBST(TreeNode* root) {
        int sum = 0;
        solve(root, sum);
        return root;
    }
private:
    // Daily
    void solve(TreeNode* root, int& sum) {
        if(root == nullptr)
            return;
        
        solve(root -> right, sum);
        
        sum += root -> val;
        root -> val = sum;
        
        solve(root -> left, sum);
    }
};